src/test/java/com/google/firebase/database/FirebaseDatabaseTest.java

```diff
@@ -44,13 +48,13 @@
           .setDatabaseUrl("https://firebase-db-test.firebaseio.com")
           .build();
 
-  @BeforeClass
-  public static void setupClass() {
+  @Before
+  public void setupTestMethod() {
     FirebaseApp.initializeApp(firebaseOptions);
```

Comment:
If you remove this line and instead move it to the test cases that need it, can you get rid of the special logic in the new tests that detects whether the default app exists?

Reply:
This adds boilerplate to a lot of tests, but makes the development experience much better while modifying tests. :+1: